sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -939,6 +939,30 @@ def predict(self, X):
 
         return self._label_binarizer.inverse_transform(y_pred)
 
+    def fit(self, X, y):
+        """Fit the model to data matrix X and target(s) y.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape (n_samples, n_features)
+            The input data.
+
+        y : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels in classification, real numbers in
+            regression).
+
+        Returns
+        -------
+        self : returns a trained MLP model.
+        """
+        if getattr(self, 'classes_', None) is not None:
+            num_classes = len(unique_labels(y))
+            if num_classes != len(self.classes_):
```

Comment:
This is not a sufficient condition, really. What if they have different labels? I think what you want to do is change the condition [here](https://github.com/scikit-learn/scikit-learn/pull/8035/files#diff-0f0be568a3c3c8f729f42b776d06243bL907) to be `if not incremental or (self.warm_start and not hasattr(self, 'classes_'))`.

It would certainly be nice if we had some helpers to make writing this logic less error-prone :\